### DISCUSSION TITLE
Confused on which Slack App token I should use

### DISCUSSION THREAD

**Discussion by simonfalke-01** [jef/streetmerchant#2436] [Help]
I am in the api.slack.com page and I see a lot of stuff. There is App ID, Client ID, Client Secret, Signing Secret, Verification token, App-level tokens, and OAuth Tokens. Which token should I use for the CAPTCHA_HANDLER_TOKEN variable in the dotenv file???
And I see under info it says "The app will need chat:write, im:history, im:write, and reactions:write permissions."
How do I configure that???

**Comment by simonfalke-01** [comment]
Running `npm run test:captcha`
gives output:

```
[2:56:45 PM] error :: ✖ couldn't get captcha response Cannot read property 'channel' of undefined {
  "stack": "TypeError: Cannot read property 'channel' of undefined\n    at Timeout._onTimeout (/Users/brandonli/Downloads/streetmerchant-main/build/src/messaging/slack.js:77:35)\n    at listOnTimeout (node:internal/timers:557:17)\n    at processTimers (node:internal/timers:500:7)"
}

```

**Reply by aellch** [reply]
Just as an FYI I'm getting the same problem -- and I made sure when I 'installed' the app from the web GUI I selected a channel, so I'm at a loss here as well.

**Comment by klords** [comment]
I had no idea this forum even existed, so sorry for the delay. Probably better to ask in the #help channel of Discord going forward. At any rate, you define the permissions in the Permissions page (see screenshot). Once you've defined the permissions, install the bot to your workspace and then copy the bot user Oauth token, which you can find at the top of the same permissions page. FYI, if you change permissions, the oauth token will change and you will have to reinstall the bot to your workspace.
![image](https://user-images.githubusercontent.com/8896801/117917478-0a297c80-b29e-11eb-941f-db8b24e26baf.png)
